### PR TITLE
Uart init

### LINKED ISSUE #1195: Improve uart-init for chips other than ESP32-C3
In #1156 the initialization of the UART peripheral was improved for C3. We need to have it for all chips to avoid inconsistency

- [ ] ESP32
- [ ] ESP32-C2
- [x] ESP32-C3
- [ ] ESP32-C6
- [ ] ESP32-H2
- [ ] ESP32-P4
- [ ] ESP32-S2
- [ ] ESP32-S3

### PR DESCRIPTION
**Pull request by bjoernQ**
Fixes #1195 - doesn't include P4 since we don't have UART for P4, yet. Should be part of the driver implementation for that chip

### CODE REVIEW

**Comment by bjoernQ** [comment]
Clippy fails with because of #1210

File: esp-hal/src/uart.rs [hunk 1]
```diff
@@ -915,47 +916,78 @@ where
         // initialize peripheral by setting clk_enable and clearing uart_reset bits
         T::enable_peripheral();
 
+        // don't reset the console UART - this will cause trouble at least on ESP32-S3
```
Comment (MabezDev) [inline esp-hal/src/uart.rs:919]: Mainly for my context, would it be possible to explain when it _would_ be okay to reset the uart being as the console? For instance, if we make it configurable later, why would someone want to enable that config option later?

**Review by MabezDev — CHANGES_REQUESTED** [review]
Thanks for looking into this! I have a couple of comments regarding maintainability but those should be easy to address.

I will also do a couple of tests with the flasher stub to make sure we didn't break anything there.

**Comment by bjoernQ** [comment]
> I will also do a couple of tests with the flasher stub to make sure we didn't break anything there.

Oh, that's good - somehow forgot about the flasher stub

**Review by MabezDev — APPROVED** [review]
Thanks for the fix, and thanks for addressing the comments! I tested in flasher stub and all looks good there (will send a esp-hal upgrade PR shortly).

**Pull request merged**